### PR TITLE
DRILL-7073: CREATE SCHEMA command / TupleSchema / ColumnMetadata improvements

### PR DESCRIPTION
**Pull request by arina-ielchiieva**
1. Add format, default, column properties logic.
2. Changed schema JSON after serialization.
3. Added appropriate unit tests.

Details in [DRILL-7073](https://issues.apache.org/jira/browse/DRILL-7073).

### CODE REVIEW

**Comment by arina-ielchiieva** [comment]
@vvysotskyi / @paul-rogers please review.

File: exec/java-exec/src/main/java/org/apache/drill/exec/record/metadata/PrimitiveColumnMetadata.java [hunk 1]
```diff
@@ -142,6 +165,87 @@ public void setDefaultValue(Object value) {
   @Override
   public Object defaultValue() { return defaultValue; }
 
+  @Override
+  public void setDefaultFromString(String value) {
```
Comment (paul-rogers) [inline exec/java-exec/src/main/java/org/apache/drill/exec/record/metadata/PrimitiveColumnMetadata.java]: This is such a handy operation that you might want to split it into two:

Object valueFromString(String value)

And your current function, which would call the above function internally.

File: exec/java-exec/src/main/java/org/apache/drill/exec/record/metadata/PrimitiveColumnMetadata.java [hunk 2]
```diff
@@ -142,6 +165,87 @@ public void setDefaultValue(Object value) {
...
+    if (value == null) {
+      return;
```
Comment (paul-rogers) [inline exec/java-exec/src/main/java/org/apache/drill/exec/record/metadata/PrimitiveColumnMetadata.java]: Set the defaulValue to null in this case?

File: exec/java-exec/src/main/java/org/apache/drill/exec/record/metadata/PrimitiveColumnMetadata.java [hunk 3]
```diff
@@ -142,6 +165,87 @@ public void setDefaultValue(Object value) {
...
+    }
+    Object objectValue = null;
+    try {
+      switch (type) {
+        case INT:
+          objectValue = Integer.parseInt(value);
```
Comment (paul-rogers) [inline exec/java-exec/src/main/java/org/apache/drill/exec/record/metadata/PrimitiveColumnMetadata.java]: Nit: can't we just say:

defaultValue = Integer.parseInt(value);
return;

File: exec/java-exec/src/main/java/org/apache/drill/exec/record/metadata/PrimitiveColumnMetadata.java [hunk 4]
```diff
@@ -142,6 +165,87 @@ public void setDefaultValue(Object value) {
...
+        case TIMESTAMP:
+          DateTimeFormatter dateTimeFormatter = formatValue == null
+            ? DateTimeFormatter.ISO_DATE_TIME.withZone(ZoneOffset.UTC) : DateTimeFormatter.ofPattern(formatValue);
+          objectValue = ZonedDateTime.parse(value, dateTimeFormatter);
+          break;
+        case INTERVAL:
+        case INTERVALDAY:
+        case INTERVALYEAR:
+          objectValue = Period.parse(value);
+          break;
```
Comment (paul-rogers) [inline exec/java-exec/src/main/java/org/apache/drill/exec/record/metadata/PrimitiveColumnMetadata.java]: missing default that throws an error for unknown type.

File: exec/java-exec/src/main/java/org/apache/drill/exec/record/metadata/PrimitiveColumnMetadata.java [hunk 5]
```diff
@@ -142,6 +165,87 @@ public void setDefaultValue(Object value) {
...
+      }
+    } catch (IllegalArgumentException | DateTimeParseException e) {
+      logger.warn("Error while parsing type {} default value {}", type, value, e);
+    }
+
+    this.defaultValue = objectValue;
+  }
+
+  @Override
+  public String defaultStringValue() {
```
Comment (paul-rogers) [inline exec/java-exec/src/main/java/org/apache/drill/exec/record/metadata/PrimitiveColumnMetadata.java]: As above, perhaps split into two functions.

File: exec/java-exec/src/main/java/org/apache/drill/exec/record/metadata/TupleSchema.java [hunk 6]
```diff
@@ -17,27 +17,52 @@
...
+@JsonInclude(JsonInclude.Include.NON_DEFAULT)
+@JsonPropertyOrder({"columns", "properties"})
 public class TupleSchema implements TupleMetadata {
 
   private MapColumnMetadata parentMap;
   private final TupleNameSpace<ColumnMetadata> nameSpace = new TupleNameSpace<>();
+  private final Map<String, String> properties = new HashMap<>();
+
+  public TupleSchema() {
+
```
Comment (paul-rogers) [inline exec/java-exec/src/main/java/org/apache/drill/exec/record/metadata/TupleSchema.java]: Nit: remove newline

File: exec/vector/src/main/java/org/apache/drill/exec/record/metadata/ColumnMetadata.java [hunk 7]
```diff
@@ -182,6 +184,10 @@
 
   int expectedElementCount();
 
+  void setFormatValue(String value);
```
Comment (paul-rogers) [inline exec/vector/src/main/java/org/apache/drill/exec/record/metadata/ColumnMetadata.java]: Perhaps I missed it, but I didn't see unit tests for this new functionality. I heavily rely on unit tests to catch regressions.

File: exec/java-exec/src/main/java/org/apache/drill/exec/record/metadata/AbstractColumnMetadata.java [hunk 8]
```diff
@@ -55,6 +72,21 @@
    */
 
   protected int expectedElementCount = 1;
+  protected Map<String, String> properties = new HashMap<>();
```
Comment (vvysotskyi) [inline exec/java-exec/src/main/java/org/apache/drill/exec/record/metadata/AbstractColumnMetadata.java]: This field can be made final. Also, In some places values from `LinkedHashMap` are put into this map. Should it be also `LinkedHashMap`?

File: exec/java-exec/src/main/java/org/apache/drill/exec/record/metadata/PrimitiveColumnMetadata.java (hunk 4 above)
Comment (arina-ielchiieva) [reply exec/java-exec/src/main/java/org/apache/drill/exec/record/metadata/PrimitiveColumnMetadata.java]: Updated the code, though when we encounter unsupported type for default value, I just log warning and return null.

File: exec/java-exec/src/main/java/org/apache/drill/exec/record/metadata/PrimitiveColumnMetadata.java (hunk 5 above)
Comment (arina-ielchiieva) [reply exec/java-exec/src/main/java/org/apache/drill/exec/record/metadata/PrimitiveColumnMetadata.java]: Done.

File: exec/vector/src/main/java/org/apache/drill/exec/record/metadata/ColumnMetadata.java (hunk 7 above)
Comment (arina-ielchiieva) [reply exec/vector/src/main/java/org/apache/drill/exec/record/metadata/ColumnMetadata.java]: I have written tests in the context schema parsing and creation.

**Comment by arina-ielchiieva** [comment]
@paul-rogers / @vvysotskyi thanks for the code review. Addressed code review comments in separate commit.

**Pull request closed without merging**